2_process/calc_doy_means.py

```diff
@@ -0,0 +1,28 @@
+import pandas as pd
+import os
+
+def read_pred_csv(csv_path):
+    df = pd.read_csv(csv_path,
+                     parse_dates = ['date'],
+                     infer_datetime_format = True)
+    return df
+
+def calc_doy_means(df, site_id, out_file=None):
+    doy_means = df.groupby([df.date.dt.month, df.date.dt.day]).mean()
+    doy_means = doy_means.reset_index(drop=True)
+    doy_means.index.name = "doy"
+    doy_means["site_id"] = site_id
+    if out_file:
+        doy_means.to_csv(out_file)
+    return doy_means
+
+def main():
```

Comment:
Like in `sb_fetch.py`, we can create the output directory if it doesn't exist yet to avoid a crash:
```suggestion
def main(out_dir):
    if not os.path.exists(out_dir):
        os.makedirs(out_dir)
```
where `out_dir` is the path to `2_process/out`. Of course, there will be other arguments besides `out_dir` passed to `main()` as well - this suggestion is just to illustrate the point.